### PR TITLE
Added Mock GoogleTest folder for PolicyLibCommon

### PR DESCRIPTION
**Pull request by v-bhavanisu**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description
Added Mock GoogleTest folder for PolicyLibCommon

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Added this mock lib definition and declaration to one of the library under Gen 11 and made sure local build is successful

## Integration Instructions

N/A

### CODE REVIEW

**Review by VivianNK — APPROVED** [review]

File: PolicyServicePkg/Test/Mock/Library/GoogleTest/MockPolicyLibCommon/MockPolicyLibCommon.inf [hunk 1]
```diff
@@ -0,0 +1,33 @@
+## @file
+# Google Test mocks for PolicyLibCommon
+#
+# Copyright (c) Microsoft Corporation.
+# Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
Comment (apop5) [inline PolicyServicePkg/Test/Mock/Library/GoogleTest/MockPolicyLibCommon/MockPolicyLibCommon.inf]:
```suggestion
#  SPDX-License-Identifier: BSD-2-Clause-Patent
```

File: PolicyServicePkg/Test/Mock/Library/GoogleTest/MockPolicyLibCommon/MockPolicyLibCommon.cpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/** @file
+  Google Test mocks for PolicyLibCommon
+
+  Copyright (c) Microsoft Corporation.
+  Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
Comment (apop5) [inline PolicyServicePkg/Test/Mock/Library/GoogleTest/MockPolicyLibCommon/MockPolicyLibCommon.cpp]:
```suggestion
  SPDX-License-Identifier: BSD-2-Clause-Patent
```

File: PolicyServicePkg/Test/Mock/Include/GoogleTest/Library/MockPolicyLibCommon.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/** @file
+  Google Test mocks for PolicyLibCommon
+
+  Copyright (c) Microsoft Corporation.
+  Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
Comment (apop5) [inline PolicyServicePkg/Test/Mock/Include/GoogleTest/Library/MockPolicyLibCommon.h]:
```suggestion
  SPDX-License-Identifier: BSD-2-Clause-Patent
```

**Review by apop5 — APPROVED** [review]

File: PolicyServicePkg/Test/Mock/Library/GoogleTest/MockPolicyLibCommon/MockPolicyLibCommon.cpp (hunk 2 above)
Comment (v-bhavanisu) [reply PolicyServicePkg/Test/Mock/Library/GoogleTest/MockPolicyLibCommon/MockPolicyLibCommon.cpp]: Copyright information corrected

File: PolicyServicePkg/Test/Mock/Include/GoogleTest/Library/MockPolicyLibCommon.h (hunk 3 above)
Comment (v-bhavanisu) [reply PolicyServicePkg/Test/Mock/Include/GoogleTest/Library/MockPolicyLibCommon.h]: Copyright information corrected

**Pull request merged**